src/Resources/ResourceFactory.php

```diff
@@ -223,6 +223,24 @@ protected function shipmentFactory(array &$data): Shipment
             unset($data['attributes']['recipient_tax_identification_numbers']);
         }
 
+        if (isset($data['relationships']['service_options'])) {
```

Comment:
What happens if `$data` has no `relationships` property?

Reply:
`isset()` catches undefined array keys, so it will skip it:

```sh
$ mp artisan tinker

> $data = []
= []

> isset($data['relationships']['service_options'])
= false
```